Tests/NIOTests/CodecTest.swift

```diff
@@ -377,4 +377,76 @@ public class MessageToByteEncoderTest: XCTestCase {
         XCTAssertFalse(try channel.finish())
 
     }
+    
+    struct fixedLengthFrameDecoderHelper<T: FixedWidthInteger> {
+        static fileprivate func fixedLengthFrameHelper(_ channel: EmbeddedChannel) throws {
+            _ = try channel.pipeline.add(handler: FixedLengthFrameDecoder<T>()).wait()
+            
+            let testStrings = ["try", "swift", "san", "jose"]
+            var buffer = channel.allocator.buffer(capacity:1024)
+            for testString in testStrings {
+                buffer.write(integer: T(testString.count))
+                buffer.write(string: testString)
+            }
+            
+            for testString in testStrings {
+                channel.pipeline.fireChannelRead(NIOAny(buffer))
```

Comment:
use `channel.writeInbound(...)` and assert the return value to replace `fireChannelRead` calls.